media-sound/mixxx/mixxx-9999.ebuild

```diff
@@ -1,127 +1,180 @@
 SLOT="0"
-IUSE="aac doc ffmpeg hid keyfinder lv2 mp3 mp4 opus qtkeychain shout wavpack"
+IUSE="aac ffmpeg keyfinder lv2 midi modplug mp3 mp4 opus qml qtkeychain"
+IUSE+=" rubberband shout test upower wavpack"
+RESTRICT="!test? ( test )"
+REQUIRED_USE="
+	qtkeychain? ( shout )
+	test? ( aac ffmpeg midi mp3 opus rubberband )
+"
 
 RDEPEND="
-	dev-db/sqlite
-	dev-libs/glib:2
+	dev-db/sqlite:3
+	dev-libs/hidapi
 	dev-libs/protobuf:=
-	dev-qt/qtconcurrent:5
-	dev-qt/qtcore:5
-	dev-qt/qtdbus:5
-	dev-qt/qtgui:5
-	dev-qt/qtnetwork:5
-	dev-qt/qtopengl:5
-	dev-qt/qtscript:5[scripttools]
-	dev-qt/qtsql:5
-	dev-qt/qtsvg:5
-	dev-qt/qtwidgets:5
-	dev-qt/qtx11extras:5
-	dev-qt/qtxml:5
-	media-libs/chromaprint
+	dev-qt/qt5compat:6
+	dev-qt/qtbase:6[concurrent,dbus,gui,icu,network,opengl,sql,sqlite,ssl,widgets,xml,X]
+	dev-qt/qtdeclarative:6
+	dev-qt/qtshadertools:6
+	dev-qt/qtsvg:6
+	media-libs/chromaprint:=
 	media-libs/flac:=
-	media-libs/libebur128
-	media-libs/libid3tag:=
+	media-libs/libebur128:=
+	media-libs/libglvnd[X]
 	media-libs/libogg
 	media-libs/libsndfile
-	media-libs/libsoundtouch
+	media-libs/libsoundtouch:=
 	media-libs/libvorbis
 	media-libs/portaudio
-	media-libs/portmidi
-	media-libs/rubberband
-	media-libs/taglib:=
-	media-libs/vamp-plugin-sdk
-	sci-libs/fftw:3.0=
-	sys-power/upower
+	<media-libs/taglib-2
+	media-sound/lame
 	virtual/glu
 	virtual/libusb:1
-	virtual/opengl
 	virtual/udev
 	x11-libs/libX11
 	aac? (
 		media-libs/faad2
 		media-libs/libmp4v2
 	)
 	ffmpeg? ( media-video/ffmpeg:= )
-	hid? ( dev-libs/hidapi )
 	keyfinder? ( media-libs/libkeyfinder )
 	lv2? ( media-libs/lilv )
-	mp3? ( media-libs/libmad )
-	mp4? ( media-libs/libmp4v2:= )
-	opus? (	media-libs/opusfile )
-	qtkeychain? ( dev-libs/qtkeychain:=[qt5(+)] )
-	shout? ( >=media-libs/libshout-2.4.5 )
+	midi? ( media-libs/portmidi )
+	modplug? ( media-libs/libmodplug )
+	mp3? (
+		media-libs/libid3tag:=
+		media-libs/libmad
+	)
+	mp4? ( media-libs/libmp4v2 )
+	opus? (
+		media-libs/opus
+		media-libs/opusfile
+	)
+	qtkeychain? ( dev-libs/qtkeychain:=[qt6(-)] )
+	rubberband? ( media-libs/rubberband:= )
+	shout? ( dev-libs/openssl:= )
+	upower? (
+		dev-libs/glib:2
+		sys-power/upower:=
+	)
 	wavpack? ( media-sound/wavpack )
 "
 DEPEND="${RDEPEND}
 	dev-cpp/ms-gsl
+	test? ( dev-cpp/gtest )
 "
 BDEPEND="
-	dev-qt/qttest:5
-	dev-qt/qtxmlpatterns:5
+	dev-util/spirv-tools
 	virtual/pkgconfig
 "
+# Not todo to prevent links
+#benchmark? (
+#	dev-cpp/benchmark
+#	dev-util/google-perftools
+#)
 
 PATCHES=(
-	"${FILESDIR}"/${PN}-9999-docs.patch
+	# Building mixxx-test target only with explicit test useflag
+	"${FILESDIR}"/${P}-tests.patch
+	# Fix strict-aliasing violations in vendored katai_cpp_stl_runtime
+	# https://github.com/kaitai-io/kaitai_struct_cpp_stl_runtime/commit/c01f530.patch
+	"${FILESDIR}"/${PN}-2.5.0-fix-strict-aliasing-kaitai.patch
 )
 
+CMAKE_SKIP_TESTS=(
+	# need HID controller
+	LegacyControllerMappingValidationTest.HidMappingsValid
+)
+
+PLOCALES="
```

Comment:
We have in general been moving away from micro-managing small files, I am not terribly excited whenever I see plocale.eclass in use.

Reply:
Especially if manuals are out.